### PR TITLE
 Unify ruby launcher finding code

### PR DESCRIPTION
**Pull request by pitr-ch**
fixes example/server.rb in GraalVM

### CODE REVIEW

**Comment by pitr-ch** [comment]
It's reformatted to match conventional common formatting of rb files.
Following 
```
rubylibprefix = \		 
expanded['rubylibprefix'] = "#{libdir}/#{ruby_base_name}"
```
is quite uncommon. It looks quite bad in IDE and cannot be auto formatted. 
The values for CONFIG and MAKEFILE_CONFIG remain grouped, the symmetry remains visible. Slight miss alignment is a small drawback compared to having to track all the local variable aliases.

**Comment by eregon** [comment]
> It's reformatted to match conventional common formatting of rb files.

No, it's IntelliJ's formatting which is not what most of our Ruby files match (notably the variable `=` alignment and other useless formatting changes).

> a small drawback compared to having to track all the local variable aliases.

Tracking local variables aliases is trivial since they have the same names as in the CONFIG/MAKEFILE_CONFIG hashes.

This introduces extra Hash lookups during startup and makes it easier to modify one value but not the other. It is not acceptable to me and this big formatting change does not belong to this PR.

**Comment by pitr-ch** [comment]
I am moving the formatting part to another PR, because I need this merged. @eregon, However note that it's perfectly fine being here. At the beginning we agreed that we won't create PR for every little thing (to safe time otherwise spend on branch and PR juggling). This is work on the same file and one change let to another, the 2 changes are related.

File: src/main/ruby/core/rbconfig.rb [hunk 1]
```diff
@@ -160,8 +160,14 @@ module RbConfig
   end
 
   def self.ruby
-    Truffle::Boot.ruby_launcher or
-        raise "we can't find the TruffleRuby launcher - set -Xlauncher=, or your launcher should be doing this for you"
+    @ruby_launcher ||= begin
+      # ruby launcher is properly set
+      Truffle::Boot.ruby_launcher ||
+          # determine launcher from ruby_home
+          ("#{CONFIG['bindir']}/#{CONFIG['ruby_install_name']}" if Truffle::Boot.ruby_home) ||
```
Comment (eregon) [inline src/main/ruby/core/rbconfig.rb]: Would this be clearer as:
```ruby
(CONFIG['bindir'] && "#{CONFIG['bindir']}/#{CONFIG['ruby_install_name']}") ||
```
?

File: src/main/ruby/core/rbconfig.rb [hunk 2]
```diff
@@ -160,8 +160,14 @@ module RbConfig
...
+          # use fallback, e.g. there is no launcher if TruffleRuby is embedded with the polyglot API
+          CONFIG['ruby_install_name']
+    end
```
Comment (eregon) [inline src/main/ruby/core/rbconfig.rb]: Why is the first element of the `||` indented differently than the rest?

File: src/main/ruby/core/rbconfig.rb (hunk 1 above)
Comment (pitr-ch) [reply src/main/ruby/core/rbconfig.rb]: I don't think so, It hides what it really depends on, which is `Truffle::Boot.ruby_home` being present.

File: src/main/ruby/core/rbconfig.rb (hunk 2 above)
Comment (pitr-ch) [reply src/main/ruby/core/rbconfig.rb]: It's one continuing line, so the 4 space rule for indentation of the continuing line applies.

File: src/main/ruby/core/rbconfig.rb (hunk 1 above)
Comment (eregon) [reply src/main/ruby/core/rbconfig.rb]: Right. I also meant the `&&` vs `if` which seems a bit odd in a conditional chain like this.

**Pull request merged**